Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -0,0 +1,306 @@
+
+    /// <summary>
+    /// Adds 0..N <see cref="BusinessObjectValidationFailure"/>s that belong to the <see cref="BocList"/> as a whole.
+    /// </summary>
+    public void AddListValidationFailures (IEnumerable<BusinessObjectValidationFailure> validationFailures)
+    {
+      ArgumentUtility.CheckNotNull(nameof(validationFailures), validationFailures);
+
+      _listValidationFailures ??= new List<HandleableFailure>();
+      _listValidationFailures.AddRange(validationFailures.Select(e => new HandleableFailure(e)));
+    }
+
+    /// <summary>
+    /// Adds 0..N <see cref="BusinessObjectValidationFailure"/>s that belong to an <see cref="IBusinessObject"/> row inside the <see cref="BocList"/>.
+    /// </summary>
+    public void AddRowValidationFailures (IBusinessObject rowObject, IEnumerable<BusinessObjectValidationFailure> validationFailures)
+    {
+      ArgumentUtility.CheckNotNull(nameof(validationFailures), validationFailures);
+
+      var failureContainer = GetOrCreateRowFailureContainer(rowObject);
+      failureContainer.AddRowFailures(validationFailures);
+    }
+
+    /// <summary>
+    /// Adds 0..N <see cref="BusinessObjectValidationFailure"/>s that belong to an <see cref="IBocColumnDefinitionWithValidationSupport"/> cell
+    /// inside an <see cref="IBusinessObject"/> row inside the <see cref="BocList"/>.
+    /// </summary>
+    public void AddCellValidationFailures (IBusinessObject rowObject, IBocColumnDefinitionWithValidationSupport cellDefinition, IEnumerable<BusinessObjectValidationFailure> validationFailures)
+    {
+      ArgumentUtility.CheckNotNull(nameof(validationFailures), validationFailures);
+
+      var failures = GetOrCreateRowFailureContainer(rowObject);
+      failures.AddCellFailures(cellDefinition, validationFailures);
+    }
+
+    /// <summary>
+    /// Returns a collection of all <see cref="BusinessObjectValidationFailure"/>s belonging to a <see cref="BocList"/>, all of
+    /// its child <see cref="IBusinessObject"/> rows and
+    /// all of the <see cref="IBocColumnDefinitionWithValidationSupport"/> cells inside those rows.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledListValidationFailures (bool markAsHandled)
+    {
+      if (_listValidationFailures == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = new List<BusinessObjectValidationFailure>();
+      foreach (var handleableRowFailure in _listValidationFailures.Where(e => !e.Handled))
+      {
+        failures.Add(handleableRowFailure.Failure);
+        if (markAsHandled)
+          handleableRowFailure.MarkAsHandled();
+      }
+
+      return failures;
+    }
+
+    /// <summary>
+    /// Returns a collection of all <see cref="BusinessObjectValidationFailure"/>s belonging to a <see cref="BocList"/>.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetAllUnhandledListValidationFailures (bool markAsHandled)
```

Comment:
What is difference between `GetUnhandledListValidationFailures` and `GetAllUnhandledListValidationFailures`. Docs say, only one of them has cell and row failures included? -> we need better naming.